src/qiskit_qir/translate.py

```diff
@@ -53,13 +52,36 @@ def to_qir_bitcode(
     :Keyword Arguments:
         * *record_output* (``bool``) --
           Whether to record output calls for registers, default `True`
-        * *use_static_qubit_alloc* (``bool``) --
-          Whether to use static qubit allocation, default `True`
-        * *use_static_result_alloc* (``bool``) --
-          Whether to use static result allocation, default `True`
     """
 
-    module = QiskitModule.from_quantum_circuit(circuit=circuit)
-    visitor = BasicQisVisitor(profile, kwargs=kwargs)
-    module.accept(visitor)
-    return visitor.bitcode()
+    module = _build_module(circuits, profile, **kwargs)
+    return module.bitcode
+
+
+def _build_module(
+    circuits: Union[QuantumCircuit, List[QuantumCircuit]],
+    profile: str = "AdaptiveExecution",
+    **kwargs
+) -> Module:
+    name = "batch"
+    if isinstance(circuits, QuantumCircuit):
+        name = circuits.name
+        circuits = [circuits]
+
+    if len(circuits) == 0:
+        raise "No QuantumCircuits provided"
+
+    llvm_module = Module(Context(), name)
+    modules = list(
+        map(
+            lambda circuit: QiskitModule.from_quantum_circuit(
+                circuit=circuit, module=llvm_module
+            ),
+            circuits,
+        )
+    )
+    for module in modules:
+        visitor = BasicQisVisitor(profile, **kwargs)
+        module.accept(visitor)
```

Comment:
nit: Wouldn't this be simpler?

```suggestion
    for circuit in circuits:
        module = QiskitModule.from_quantum_circuit(circuit, llvm_module)
        module.accept(BasicQisVisitor(profile, **kwargs))
```